app/models/deploy_service.rb

```diff
@@ -19,7 +19,13 @@ def deploy!(stage, attributes)
   def confirm_deploy!(deploy)
     send_before_notifications(deploy)
 
-    job_execution = JobExecution.start_job(deploy.reference, deploy.job)
+    stage = deploy.stage
+
+    job_execution = JobExecution.start_job(
+      deploy.reference, deploy.job,
+      construct_env(stage).merge(key: stage.id)
```

Comment:
Any reason why you merge `key` here, rather than including it in `construct_env`?